api/endpoints/skill.py

```diff
@@ -215,6 +216,39 @@ async def list_sub_skills(*, root_skill_id: str, user: User | None = public_auth
     )
 
 
+@router.get("/skilltree/subskills/search", responses=responses(SearchResults))
+@redis_cached("skills", "search_term", "user")
+async def search_sub_skill(*, search_term: str, user: User | None = public_auth) -> Any:
+    """Search for sub skills, root skills, and courses."""
+
+    sub_skills = [
+        sub_skill.serialize
+        async for sub_skill in await db.stream(
+            select(models.SubSkill).where(
+                (models.SubSkill.id.ilike(f"%{search_term}%")) | (models.SubSkill.name.ilike(f"%{search_term}%"))
```

Comment:
Try something like [this](https://github.com/Bootstrap-Academy/auth-ms/blob/d0dc6d00dd77ed1aa05ec7ecfc2d2725175bc500/api/endpoints/user.py#L82-L87) instead. Also, don't search in the skill id, this could just be a random uuid